dkg/nodesigs.go

```diff
@@ -85,21 +129,53 @@ func (n *nodeSigBcast) exchange(
 	}
 
 	ret := make([][]byte, n.operatorAmt)
-	ret[nodeIdx.PeerIdx] = sig
+	ret[n.nodeIdx.PeerIdx] = sig
+
+	for {
+		var done bool
 
-	// see newLockHashK1Bcast comment
-	for idx := 0; idx < n.operatorAmt-1; idx++ {
 		select {
 		case <-ctx.Done():
-			return nil, errors.New("context canceled")
-		case otherSig := <-n.otherSigs:
+			return nil, ctx.Err()
+		default:
+			// bcast returns len(peers)-1 messages to each peer, so that senders don't get their own message
+			// hence wait for operatorAmt-1 messages
+			done = mapLen(&n.otherSigs) == n.operatorAmt-1
```

Comment:
why not store `localSig` in the `n.otherSigs` then it follows the same flow as other sigs and no comments are required?